scripts/integration-tests/e2e-jest.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+#==============================================================================#
+#                                  SETUP                                       #
+#==============================================================================#
+
+# Start in scripts/integration-tests/ even if run from root directory
+cd "$(dirname "$0")" || exit
+root="$PWD"
+
+source utils/local-registry.sh
+source utils/cleanup.sh
+
+# Echo every command being executed
+set -x
+
+# Clone jest
+git clone --depth=1 https://github.com/facebook/jest /tmp/jest
+cd /tmp/jest || exit
+
+# Update @babel/* dependencies
+bump_deps="$root/utils/bump-babel-dependencies.js"
+node "$bump_deps"
+for d in ./packages/*/
+do
+  (cd "$d"; node "$bump_deps")
+done
+
+#==============================================================================#
+#                                 ENVIRONMENT                                  #
+#==============================================================================#
+node -v
+yarn --version
+python --version
+
+#==============================================================================#
+#                                   TEST                                       #
+#==============================================================================#
+
+startLocalRegistry "$root"/verdaccio-config.yml
+yarn install
+yarn dedupe '@babel/*'
+yarn build
+
+# The full test suite takes about 20mins on CircleCI. We run only a few of them
+# to speed it up.
+# The goals of this e2e test are:
+#   1) Check that the typescript compilation isn't completely broken
+#   2) Make sure that we don't accidentally break jest's usage of the Babel API
+CI=true yarn test-ci-partial packages
+CI=true yarn test-ci-partial e2e/__tests__/babel
+CI=true yarn test-ci-partial e2e/__tests__/transform
```

Comment:
Could do `yarn test-ci-partial --ci packages e2e/__tests__/babel e2e/__tests__/transform`?

FWIW I just landed https://github.com/facebook/jest/pull/10653 which adds `test-ci-partial:parallel` (which just removes the `-i` argument) so if you have multiple cores available on CI you can speed it up.

`yarn test-ci-partial --ci --max-workers $(nproc) packages e2e/__tests__/babel e2e/__tests__/transform` or something like that